### PR TITLE
Feature/m UI c 243 chat restore

### PR DESCRIPTION
**Pull request by eigom**
https://salemove.atlassian.net/browse/MUIC-243

- When tapping Chat when engagement was active resulted in a new engagement window to be created and `Salemove.sharedInstance.queueForEngagement()` to be called again. In this implementation this will not happen - a warning is printed to the console and tapping on Chat, Audio call etc won't do anything. You get back to Glia only via bubble. To test if engagement is re-established one has to kill the app while session is active.
- Since the SDK uses a singleton (`Salemove.sharedInstance`), there can be only one widgets instance and thus Glia is now also a singleton
- Re-worked and extended events sent to integrator's app

### CODE REVIEW

**Comment by VladKupriienko** [comment]
Sorry for the failed Bitrise check. Forgot to remove the trigger while it's not done.

**Review by VladKupriienko — APPROVED** [review]

**Review by gersonnoboa — COMMENTED** [review]
I tested it by 

1. Starting an engagement, then stopping the app in Xcode, pressing Play again, starting an engagement again.
2. Starting an engagement, killing the app in iOS, opening the app in the app drawer, starting an engagement again.

But with neither I could manage to send messages from operator to visitor.

File: GliaWidgets/Glia.swift [hunk 1]
```diff
@@ -19,39 +21,68 @@ public protocol SceneProvider: class {
     func windowScene() -> UIWindowScene?
 }
 
-public protocol GliaDelegate: class {
-    func event(_ event: GliaEvent)
-}
-
 public class Glia {
+    public static let shared = Glia()
```
[the remark this answers is not in the document]
Comment (eigom) [reply GliaWidgets/Glia.swift:25]: Did 10+ tests on simulator and device and had no problems. Will do some more.

File: TestingApp/ViewController.swift [hunk 2]
```diff
@@ -67,14 +67,29 @@ extension ViewController {
         let theme = settingsViewController.theme
         let visitorContext = VisitorContext(type: .page,
                                             url: "https://www.salemoveinsurance.com")
-
-        glia = Glia(configuration: conf)
+        Glia.shared.onEvent = { event in
```
[the remark this answers is not in the document]
Comment (eigom) [reply TestingApp/ViewController.swift:70]: So integrator would know what is going on if needed. To update UI for example - hide/dim engagement start button when Glia UI is presented. Or for analytics tracking.
This was not in requirements and I can remove it if you think this won't be needed.

[the remark this answers is not in the document]
Comment (gersonnoboa) [reply TestingApp/ViewController.swift:70]: Ahhh so this is like an example of how to use that. Okay that is fine.

**Comment by eigom** [comment]
Here's what I found:

- App calls `queueForEngagement()` on start and sets state to `.enqueued` on success
- App sets state to `.engaged` in `start()` callback
- If there is an ongoing engagement, then `start()` was called before call to `queueForEngagement()` has finished. So state was set to `.engaged` and then to `.enqueued`.

I've added check to prevent this. Please check if this has fixed anything for you before I'll continue investigating.

**Review by gersonnoboa — APPROVED** [review]
It works now

**Pull request merged**